sklego/linear_model.py

```diff
@@ -64,3 +68,126 @@ def predict(self, X):
+
+    :param covariance_threshold: The maximum allowed covariance between the sensitive attributes and the distance to the
+    decision boundary. If set to None, no fairness constraint is enforced
+    :param sensitive_cols: List of sensitive column names(when X is a dataframe)
+    or a list of column indices when X is a numpy array.
+    :param C: Inverse of regularization strength; must be a positive float.
+    Like in support vector machines, smaller values specify stronger regularization.
+    :param penalty: Used to specify the norm used in the penalization. Expects 'none' or 'l1'
+    :param fit_intercept: Specifies if a constant (a.k.a. bias or intercept) should be added to the decision function.
+    :param max_iter: Maximum number of iterations taken for the solvers to converge.
+    :param multi_class: The method to use for multiclass predictions
+    :param n_jobs: The amount of parallel jobs thata should be used to fit multiclass models
+
+    """
+
+    def __new__(cls, *args, multi_class='ovr', n_jobs=1, **kwargs):
+        multiclass_meta = {
+            'ovr': OneVsRestClassifier,
+            'ovo': OneVsOneClassifier,
+        }[multi_class]
+        return multiclass_meta(_FairClassifier(*args, **kwargs), n_jobs=n_jobs)
+
+
+class _FairClassifier(BaseEstimator, LinearClassifierMixin):
+    def __init__(self, covariance_threshold, sensitive_cols, C=1.0, penalty='l1', fit_intercept=True, max_iter=100):
+        self.sensitive_cols = sensitive_cols
+        self.fit_intercept = fit_intercept
+        self.penalty = penalty
+        self.covariance_threshold = covariance_threshold
+        self.max_iter = max_iter
+        self.C = C
+
+    def fit(self, X, y):
+        if self.penalty not in ['l1', 'none']:
+            raise ValueError(f"penalty should be either 'l1' or 'none', got {self.penalty}")
+
+        X, y = check_X_y(X, y, accept_large_sparse=False)
+
+        sensitive = X[:, self.sensitive_cols] if isinstance(X, np.ndarray) else X[self.sensitive_cols]
+        X = self.add_intercept(X)
+
+        column_or_1d(y)
+        label_encoder = LabelEncoder().fit(y)
+        y = label_encoder.transform(y)
+        self.classes_ = label_encoder.classes_
+
+        if len(self.classes_) > 2:
+            raise ValueError(f"This solver needs samples of exactly 2 classes"
+                             f" in the data, but the data contains {len(self.classes_)}: {self.classes_}")
+
+        self._fit(sensitive, X, y)
+        return self
+
+    def _fit(self, sensitive, X, y):
+        n_obs, n_features = X.shape
+        theta = cp.Variable(n_features)
+        y_hat = X @ theta
+
+        log_likelihood = cp.sum(
+            cp.multiply(y, y_hat) -
+            cp.log_sum_exp(cp.hstack([np.zeros((n_obs, 1)), cp.reshape(y_hat, (n_obs, 1))]), axis=1)
+        )
+        if self.penalty == 'l1':
+            log_likelihood - cp.sum((1 / self.C) * cp.norm(theta[1:]))
```

Comment:
shouldn't this be `-=` instead of `-`?

Reply:
Yes! that's why my regularization tests didn't work, thanks

Reply:
makes sense :)